.github/workflows/check-code-quality.yml

```diff
@@ -1,5 +1,5 @@
 on: pull_request
-name: Run tests
+name: Check code quality
```

Comment:
I believe this should be a separate job, one for the tests and one for the linter with proper names. In that way you can easily understand where the problem will be, and it would be easier to maintain